src/main/java/com/atlassian/bitbucket/jenkins/internal/http/HttpRequestExecutorImpl.java

```diff
@@ -30,20 +30,25 @@ public HttpRequestExecutorImpl(Call.Factory httpCallFactory) {
     @Override
     public <T> T executeGet(HttpUrl url, BitbucketCredentials credential, ResponseConsumer<T> consumer) {
         Request.Builder requestBuilder = new Request.Builder().url(url);
-        if (credential != ANONYMOUS_CREDENTIALS) {
-            requestBuilder.addHeader(AUTHORIZATION_HEADER_KEY, credential.toHeaderValue());
-        }
+        addAuthorization(credential, requestBuilder);
+        return executeRequest(requestBuilder.build(), consumer);
+    }
+
+    @Override
+    public <T> T executePost(HttpUrl url, BitbucketCredentials credential, String requestBody, ResponseConsumer<T> consumer) {
+        MediaType mediaType = MediaType.parse(requestBody);
```

Comment:
Do we want to detect the media type, considering the facts that all of the requests/responses will be `JSON` and detecting isn't exactly free? 
Do you see a use case for any other media type?

Reply:
> detecting isn't exactly free?

Reply:
Yes, other concern is that it don't adds any value. Do you have a use case for non JSON media type?

Reply:
It feels a bit odd to always assume the request body is JSON when there can be multiple data types.

Reply:
Encoding is another thing that we assume to be `UTF-8`, it's ok to assume the body is of type JSON

Reply:
`UTF-8` is cross-cutting and applies to all of the requests. JSON is not. I am happy to take a type as the parameter as input but not always assuming it to be JSON. 
Let see what others think.

Reply:
I have assumed it to be JSON